src/kernels/kernel.ts

```diff
@@ -248,20 +263,23 @@ export class Kernel implements IKernel {
         const disposeImpl = async () => {
             traceInfo(`Dispose kernel ${(this.resourceUri || this.notebookDocument.uri).toString()}`);
             this.restarting = undefined;
+            const promises: Promise<void>[] = [];
+            promises.push(this.kernelExecution.cancel());
```

Comment:
Found that when shutting down a kernel, we didn't cancel the pending executions, all we did was dispose that object.
This is kinda debt